vmas/simulator/dynamics/drone.py

```diff
@@ -0,0 +1,128 @@
+from typing import Union
+
+import torch
+import vmas.simulator.core
+import vmas.simulator.utils
+from vmas.simulator.dynamics.common import Dynamics
+
+
+class Drone(Dynamics):
```

Comment:
You need to implement the reset function

Reply:
In that function you should reset the internal state of the controller to 0s , but do not touch the vmas state.
This is why I would keep the 2 states separate and cat them only in the process_action function.

The function will take an `env_index` (look at parent class).
What you need to do is:
- when the index is None, reset all envs
- when the index is int, reset only the environment at that index

Reply:
Also, I'm thinking of the correct way to trigger the reset function. For example, the drone has flipped over when some internal state (pitch, roll) goes over some threshold. Thus, the reset() function should be called. Can this be included during training?

Reply:
the reset function is automatically handled by vmas. People can decide when to trigger it based on the scenario. For now let's not think about this

Reply:
see here for example https://github.com/proroklab/VectorizedMultiAgentSimulator/blob/a883c90248f264136cc1840c911da96c87d5a4d7/vmas/simulator/controllers/velocity_controller.py#L80

Reply:
Thanks. I think I got it. I will double check with you tomorrow in-person before updating the code.